scripts/cbmc_json_parser.py

```diff
@@ -0,0 +1,409 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+
+"""CBMC JSON Parser
+
+This script allows rmc to print to the console the final output displayed to the user
+after CBMC gives the response JSON object.
+
+This script accepts JSON files.
+
+This script requires that `colorama` be installed within the Python
+environment you are running this script in.
+
+This file can also be imported as a module and contains the following
+functions:
+
+    * transform_cbmc_output - returns the formatted string output after parsing
+    * main - the main function of the script
+"""
+
+import json
+from json.decoder import JSONDecodeError
+from os import remove
+from os.path import exists
+from colorama import Fore, Style
+import tempfile
+import sys
+from enum import Enum
+
+# Enum to store the style of output that is given by the argument flags
+output_style_switcher = {
+    'default': 'old',
+    'regular': 'regular',
+    'terse': 'terse',
+    'old': 'old'
+}
+
+class GlobalMessages(str, Enum):
+    """
+    Enum class to store all the global messages
+    """
+    CONST_TRACE_MESSAGE = 'Building error trace',
+    PROGRAM = 'program'
+    RESULT = 'result'
+    MESSAGE_TEXT = 'messageText'
+    SUCCESS = 'SUCCESS'
+    FAILED = 'FAILED'
+
+def main():
+
+    # Check only one json file as input
+    if len(sys.argv) < 2:
+        print("Json File Input Missing")
+        sys.exit(1)
+
+    # parse the input json file
+    with open(sys.argv[1]) as f:
+        sample_json_file_parsing = f.read()
+
+    # the main function should take a json file as input
+    transform_cbmc_output(sample_json_file_parsing, output_style=output_style_switcher["old"])
+    return
+
+def transform_cbmc_output(log_file, output_style):
+    """
+    Take Unstructured CBMC Response object, parse the blob and gives structured
+    and formatted output depending on User Provided Output Style
+
+    Parameters -
+        log_file : str
+            A response blob that is given to the function from CBMC
+        output_style : int
+            An index to tell the script which style of output is requested by the User.
+
+    Returns -
+        None
+            Prints the final output string to the user
+    """
+
+    # Output Message is the final output that is printed to the user
+    output_message = ""
+
+    # Check if the output given by CBMC is in Json format or not
+    is_json_bool, json_file = is_json(log_file)
+    if is_json_bool:
+
+        # Write cbmc output to a json file in scripts/temp_json for parallel I/O
+        json_file = save_json_file(json_file)
+
+        # Read from the saved json file and restructure it to make it parseable
+        parseable_json_dictionary = restructure_json_file(json_file)
+
+        # Extract property information from the restructured JSON file
+        properties, solver_information = extract_solver_information(parseable_json_dictionary)
```

Comment:
Better names needed. I'm actually unclear on what these are but if I'm guessing right then it almost seems backwards. Like, `properties` seems to be summary information from the solver while `solver_information` seems to be each individual property?